src/AMSlib/AMS.cpp

```diff
@@ -485,6 +491,8 @@ class AMSWrap
 public:
   AMSWrap() : memManager(ams::ResourceManager::getInstance())
   {
+    rId = get_rank_id();
```

Comment:
I am a little worried about this here. The `env` variables I am using are not always supported and depend on the job scheduler (on a non-workflow execution we don't have any control over). I skimmed over the code and this seems to be mostly used as an identifier in Logging messages right?

Which part of rId is necessary for instantiating the db before application execution?

Reply:
I agree this solution is not the best. The problem here is that I cannot use `MPI_Comm_rank` in RMQ part because AMSWrap is static and will get instantiated before `MPI_Init` is being called resulting an an MPI error.

I use the `rId` for three things
1. For logging
2. To embed the MPI rank in AMSMessage
3. To name uniquely inbound queues (queue that receive messages from the outside). Each of these should be uniquely named otherwise some ranks will not receive messages with the new ML model (for example).

I am working on a solution where I use MPI to get the rank before sending the `AMSMessage` (at that point MPI is initialized) and I use a random generator to uniquely named RMQ queue in the `RMQConsumerHandler`. For the logging, the `rId` will be the random number, so it's not perfect but it's okay I think.

Reply:
Ok. I will go over the implementation and see if I get any idea. (1) is not that important and for 1 you can use the current id. 
2. Should be feasible. Since the DBManager creates files in the form `data_<rid>.<suffix>`. I don't recall the details of AMSMessage. So maybe that is too late in the pipeline.
3. Can we build a lazy queue creation mechanism? That takes the first time we call the `getLatestModel` ?

Reply:
I pushed a new version without `rId` by implementing 1. and 2.

It would be complicated (but not impossible) to build a lazy queue creation mechanism. The problem is that we have to create the queue before any messages is being sent to AMSlib, otherwise we will lose these messages (if you send a message to a queue that does not exist then this message is lost).
If we know the names of the queues beforehand, we could create them on the Python side for example. If these queues exist then messages sent to them will be kept.